server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridTiler.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.bucket.geogrid;
+
+import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.geometry.utils.Geohash;
+import org.elasticsearch.index.fielddata.MultiGeoValues;
+import org.locationtech.spatial4j.io.GeohashUtils;
```

Comment:
Is this temporary?

Reply:
I left it here for now because I wasn't sure the best way to port this helper method into ES: https://github.com/locationtech/spatial4j/blob/master/src/main/java/org/locationtech/spatial4j/io/GeohashUtils.java#L187-L200

Reply:
I went ahead and ported this functionality into Geohash.java